File: java/SqlInjection.java

import java.sql.*;
public class SqlInjection {
    public void getUser(String userId) throws SQLException {
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/test", "root", "password");
        // VULNERABLE: user input directly concatenated into SQL query
        String query = "SELECT * FROM users WHERE id=" + userId; // CWE-89
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(query);
    }
}

File: go/sql_injection.go
package main
import (
    "database/sql"
    _ "github.com/go-sql-driver/mysql"
    "os"
)
func main() {
    db, _ := sql.Open("mysql", "root:password@/test")
    user := os.Args[1]
    // VULNERABLE: unsanitized input
    db.Query("SELECT * FROM users WHERE name=" + user) // CWE-89
}
